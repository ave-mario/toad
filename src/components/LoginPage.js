import React from 'react';
import { Link } from 'react-router-dom';
import Title from '../elements/Title';

function LoginPage() {
  return (
    <div>
      <Title>Login Page</Title>
      <Title color="red">Login Page</Title>
      <span>Do not have account?</span>
      <Link to="register">Register here</Link>
    </div>
  );
}

export default LoginPage;
