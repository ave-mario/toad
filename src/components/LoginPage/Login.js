/* eslint-disable react/prop-types */
import React from 'react';
import { Header, Main, Footer } from '../../elements';
import {
  SubmitButton,
  LoginForm,
  LoginInput,
  LoginTitle,
  LoginSubTitle
} from './elements/login.form';

function Login(props) {
  const {
    errors,
    touched,
    values,
    handleChange,
    handleBlur,
    handleSubmit
  } = props;
  return (
    <LoginForm onSubmit={handleSubmit}>
      <Header>
        <LoginTitle>Welcome to hotel management</LoginTitle>
        <LoginSubTitle>Sign in</LoginSubTitle>
      </Header>
      <Main>
        <LoginInput
          className="emailInput"
          type="text"
          name="email"
          placeholder="Email"
          onChange={handleChange}
          border={errors.email && touched.email && '1px solid red'}
          value={values.mail}
          onBlur={handleBlur}
        />
        {errors.email && touched.email && (
          <div style={{ color: 'red' }}>{errors.email}</div>
        )}
        <LoginInput
          className="passwordInput"
          type="password"
          name="password"
          placeholder="Password"
          onChange={handleChange}
          border={errors.password && touched.password && '1px solid red'}
          value={values.password}
          onBlur={handleBlur}
        />
        {errors.password && touched.password && (
          <div style={{ color: 'red' }}>{errors.password}</div>
        )}
      </Main>
      <Footer>
        <SubmitButton className="submitButton" type="submit">
          SIGN IN
        </SubmitButton>
      </Footer>
    </LoginForm>
  );
}

export default Login;
