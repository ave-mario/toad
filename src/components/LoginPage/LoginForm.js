import React from 'react';
import { withFormik } from 'formik';
import { withTranslation } from 'react-i18next';
import PropTypes from 'prop-types';
import { Header, Main, Footer, ErrorText } from 'elements';
import { dispatcher } from 'config/redux.store';
import authActions from 'actions/auth.actions';
import { addResource } from 'config/localize';
import { loginValidationSchema } from 'validation/auth.validation';
import ErrorBar from '../ErrorBar';
import { dispatcher } from '../../config/redux.store';
import authActions from '../../actions/auth.actions';
import { addResource } from '../../config/localize';
import { loginValidationSchema } from '../../config/validation.schemas';
import {
  SubmitButton,
  LoginForm,
  LoginInput,
  LoginTitle,
  LoginSubTitle
} from './elements/login.form';
import localization from './localize.login';

addResource('LoginForm', localization);

const { Creators } = authActions;

export const handleSubmit = values => {
  dispatcher(Creators.loginRequest(values.email, values.password));
};

export const mapPropsToValues = () => ({
  email: '',
  password: ''
});

export const MyFormInner = props => {
  const {
    values,
    handleChange,
    handleBlur,
    errors,
    touched,
    handleSubmit: submit,
    isSubmitting,
    isFailed,
    t
  } = props;
  return (
    <LoginForm onSubmit={submit}>
      <Header>
        <LoginTitle>{t('labels.title')}</LoginTitle>
        <LoginSubTitle>{t('labels.signIn')}</LoginSubTitle>
      </Header>
      <Main>
        <LoginInput
          id="emailInput"
          className="emailInput"
          type="text"
          name="email"
          placeholder={t('labels.email')}
          onChange={handleChange}
          border={errors.email && touched.email && '1px solid red'}
          value={values.email}
          onBlur={handleBlur}
        />
        {errors.email && touched.email && <ErrorText>{errors.email}</ErrorText>}
        <LoginInput
          className="passwordInput"
          type="password"
          name="password"
          placeholder={t('labels.password')}
          onChange={handleChange}
          border={errors.password && touched.password && '1px solid red'}
          value={values.password}
          onBlur={handleBlur}
        />
        {errors.password && touched.password && (
          <ErrorText>{errors.password}</ErrorText>
        )}
        <ErrorBar />
      </Main>
      <Footer>
        <SubmitButton
          className="submitButton"
          type="submit"
          disabled={isSubmitting && !isFailed}
        >
          {t('buttons.submit')}
        </SubmitButton>
      </Footer>
    </LoginForm>
  );
};

MyFormInner.propTypes = {
  values: PropTypes.objectOf(PropTypes.string).isRequired,
  handleChange: PropTypes.func.isRequired,
  handleBlur: PropTypes.func.isRequired,
  errors: PropTypes.objectOf(PropTypes.string).isRequired,
  touched: PropTypes.objectOf(PropTypes.bool).isRequired,
  handleSubmit: PropTypes.func.isRequired,
  isSubmitting: PropTypes.bool.isRequired,
  isFailed: PropTypes.bool,
  t: PropTypes.func.isRequired
};
MyFormInner.defaultProps = {
  isFailed: false
};

export const Login = withFormik({
  mapPropsToValues,
  validationSchema: loginValidationSchema,
  handleSubmit
})(withTranslation('LoginForm')(MyFormInner));
