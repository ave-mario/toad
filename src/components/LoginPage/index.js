import LoginPageContainer from './container';

export default LoginPageContainer;
