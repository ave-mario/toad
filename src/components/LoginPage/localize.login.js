export default {
  ru: {
    labels: {
      title: 'Добро пожаловать в гостиничный менеджмент',
      signIn: 'Вход',
      email: 'Email',
      password: 'Пароль'
    },
    buttons: {
      submit: 'Войти'
    }
  },
  en: {
    labels: {
      title: 'Welcome to hotel management',
      signIn: 'Sign in',
      email: 'Email',
      password: 'Password'
    },
    buttons: {
      submit: 'Sign in'
    }
  }
};
