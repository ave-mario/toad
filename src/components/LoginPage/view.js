/* eslint-disable react/no-unused-state */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Header, Main, Footer } from '../../elements';
import {
  SubmitButton,
  LoginForm,
  LoginInput,
  LoginTitle,
  LoginSubTitle
} from './elements/login.form';

class LoginPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      isLogged: false
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  handleSubmit(e) {
    const { email, password } = this.state;
    const { login } = this.props;
    if (email === 'vidgf@sdf.sfn' && password === '12345QWE') {
      this.setState({ isLogged: true });
    }
    e.preventDefault();
    login(email, password);
  }

  render() {
    return (
      <>
        <LoginForm>
          <Header>
            <LoginTitle>Welcome to hotel management</LoginTitle>
            <LoginSubTitle>Sign in</LoginSubTitle>
          </Header>
          <Main>
            <LoginInput
              className="emailInput"
              type="text"
              name="email"
              placeholder="Email"
              onChange={this.handleChange}
            />
            <LoginInput
              className="passwordInput"
              type="password"
              name="password"
              placeholder="Password"
              onChange={this.handleChange}
            />
          </Main>
          <Footer>
            <SubmitButton
              className="submitButton"
              type="submit"
              onClick={this.handleSubmit}
            >
              SIGN IN
            </SubmitButton>
          </Footer>
        </LoginForm>
      </>
    );
  }
}

LoginPage.propTypes = {
  login: PropTypes.func.isRequired
};
export default LoginPage;
