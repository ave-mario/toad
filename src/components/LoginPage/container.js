import { connect } from 'react-redux';
import login from '../../actions/auth.actions';
import LoginPage from './view';
import store from '../../store';

const mapStateToProps = state => ({
  user: state.user
});

export default connect(
  mapStateToProps,
  { login, store }
)(LoginPage);
