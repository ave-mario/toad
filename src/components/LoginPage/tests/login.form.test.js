import React from 'react';
import { shallow } from 'enzyme';
import faker from 'faker';
import LoginPage, { fakeValidEmail, fakeValidPassword } from '../view';

const randomEmail = faker.internet.email;
const randomPassword = faker.internet.password;
describe('Test case for testing login form', () => {
  let wrapper;
  const props = {
    login: jest.fn()
  };

  test('email check', () => {
    wrapper = shallow(<LoginPage {...props} />);
    wrapper.find('.emailInput').simulate('change', {
      target: { name: 'email', value: randomEmail }
    });

    expect(wrapper.state('email')).toEqual(randomEmail);
  });

  test('password check', () => {
    wrapper = shallow(<LoginPage {...props} />);
    wrapper.find('.passwordInput').simulate('change', {
      target: { name: 'password', value: randomPassword }
    });

    expect(wrapper.state('password')).toEqual(randomPassword);
  });

  test('login check with right data', () => {
    wrapper = shallow(<LoginPage {...props} />);

    wrapper.find('.emailInput').simulate('change', {
      target: { name: 'email', value: fakeValidEmail }
    });
    wrapper.find('.passwordInput').simulate('change', {
      target: { name: 'password', value: fakeValidPassword }
    });
    wrapper.find('.submitButton').simulate('click', {
      preventDefault: () => {}
    });

    expect(props.login).toBeCalled();
    expect(wrapper.state('isLogged')).toBe(true);
  });

  test('login check with wrong data', () => {
    wrapper = shallow(<LoginPage {...props} />);

    wrapper.find('.emailInput').simulate('change', {
      target: { name: 'username', value: randomEmail }
    });
    wrapper.find('.passwordInput').simulate('change', {
      target: { name: 'password', value: randomPassword }
    });
    wrapper.find('.submitButton').simulate('click', {
      preventDefault: () => {}
    });

    expect(props.login).toBeCalled();
    expect(wrapper.state('isLogged')).toBe(false);
  });
  test('Login page snapshot', () => {
    const component = shallow(<LoginPage {...props} />);

    expect(component).toMatchSnapshot();
  });
});
