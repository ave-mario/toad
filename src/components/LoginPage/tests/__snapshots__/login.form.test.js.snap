// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test case for testing login form Login page snapshot 1`] = `
<Fragment>
  <Styled(styled.form)>
    <styled.header>
      <Styled(styled.h1)>
        Welcome to hotel management
      </Styled(styled.h1)>
      <Styled(styled.h2)>
        Sign in
      </Styled(styled.h2)>
    </styled.header>
    <styled.main>
      <Styled(styled.input)
        className="emailInput"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        type="text"
      />
      <Styled(styled.input)
        className="passwordInput"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
      />
    </styled.main>
    <styled.footer>
      <Styled(styled.button)
        className="submitButton"
        onClick={[Function]}
        type="submit"
      >
        SIGN IN
      </Styled(styled.button)>
    </styled.footer>
  </Styled(styled.form)>
</Fragment>
`;
